hw/ip/spi_device/data/spi_device_testplan.hjson

```diff
@@ -146,5 +146,102 @@
       milestone: V2
       tests: []
     }
+    {
+      name: tpm_fifo_read
+      desc: '''
+            - Set TPM_CFG.TPM_MODE to 0.
+            - Set TPM_CFG.EN.
+            - Randomise other fields in TPM_CFG.
+            - Send TPM read command over the SPI bus with a randomised address.
+            - Check TPM_CMD_ADDR.
+            - Check TPM_STATUS.cmdaddr_notempty and INTR_STATE.tpm_header_notempty, they should be asserted if hw_reg_dis == 0.
+            - If hw_reg_dis == 0, the data is returned to the host via return-by-HW register, else the data is returned via read FIFO.
+            - Check TPM_CFG.INVALID_LOCALITY.
```

Comment:
this should be randomized, right? I will affect if the command and address are updated to FIFO or the data directly returns by HW.